ts/components/services/ServiceList.tsx

```diff
@@ -15,38 +32,123 @@ type AnimatedProps = {
 };
 
 type OwnProps = {
+  sections: ReadonlyArray<SectionListData<pot.Pot<ServicePublic, Error>>>;
+  profile: ProfileState;
+  isRefreshing: boolean;
+  onRefresh: () => void;
+  onSelect: (service: ServicePublic) => void;
+  readServices: ReadStateByServicesId;
+  isExperimentalFeaturesEnabled?: boolean;
   ListEmptyComponent?: React.ComponentProps<
-    typeof FlatList
+    typeof SectionList
   >["ListEmptyComponent"];
 };
 
 type Props = OwnProps & AnimatedProps;
 
-const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
+const styles = StyleSheet.create({
+  listItem: {
+    paddingLeft: variables.contentPadding,
+    paddingRight: variables.contentPadding
+  },
+  padded: {
+    marginLeft: customVariables.contentPadding,
+    marginRight: customVariables.contentPadding
+  }
+});
+
+const AnimatedSectionList = Animated.createAnimatedComponent(SectionList);
 
 class ServiceList extends React.Component<Props> {
-  private flatListRef = React.createRef<typeof AnimatedFlatList>();
+  private sectionListRef = React.createRef<typeof AnimatedSectionList>();
+
+  private renderServiceItem = (
+    itemInfo: ListRenderItemInfo<pot.Pot<ServicePublic, Error>>
+  ) => (
+    <NewServiceListItem
+      item={itemInfo.item}
+      profile={this.props.profile}
+      onSelect={this.props.onSelect}
+      isRead={this.isRead(itemInfo.item, this.props.readServices)}
+      hideSeparator={true}
+    />
+  );
+
+  private getServiceKey = (
+    potService: pot.Pot<ServicePublic, Error>,
+    index: number
+  ): string => {
+    return pot.getOrElse(
+      pot.map(
+        potService,
+        service => `${service.service_id}-${service.version}`
+      ),
+      `service-pot-${index}`
+    );
+  };
 
-  private renderItem = () => {
-    return <View />;
+  private renderServiceSectionHeader = (info: {
+    section: ServicesSectionState;
+  }): React.ReactNode => (
+    <SectionHeaderComponent
+      sectionHeader={info.section.organizationName}
+      style={styles.padded}
+      logoUri={getLogoForOrganization(info.section.organizationFiscalCode)}
+    />
+  );
+
+  private isRead = (
+    potService: pot.Pot<ServicePublic, Error>,
+    readServices: ReadStateByServicesId
+  ): boolean => {
+    const service =
```

Comment:
could be this
```typescript
const service =
      pot.isLoading(potService) ||
      pot.isError(potService) ||
      pot.isNone(potService)
        ? undefined
        : potService.value;

    return (
      readServices !== undefined &&
      service !== undefined &&
      readServices[service.service_id] !== undefined
    );
```
replaced with this?
```typescript
return pot.getOrElse(
      pot.map(potService, s => s && readServices[s.service_id] !== undefined),
      false
    );
```

Reply:
yes, updated within 3e8dd55e69564af5309be09820d64059dd82cc3c